commands/bake.go

```diff
@@ -87,7 +87,7 @@ func runBake(dockerCli command.Cli, targets []string, in bakeOptions) (err error
 	}()
 
 	dis, err := getInstanceOrDefault(ctx, dockerCli, in.builder, contextPathHash)
-	if err != nil {
+	if err != nil && (url != "" || !in.printOnly) {
```

Comment:
IIUC, in this case;

- we don't need the instance
- but we're still calling`getInstanceOrDefault()` (but ignore if it fails)

Could we make the condition so that we skip capping `getInstanceOrDefault()` if `url == ""` and `in.printOnly`? My concern would be that calling `getInstanceOrDefault()` (if the instance is not accessible) would result in a delay (buildx trying to access it, which may result in a "timeout" after XX seconds).

Reply:
> would result in a delay (buildx trying to access it, which may result in a "timeout" after XX seconds).

It might if context is set to a remote docker endpoint but not for a socket one.

> * but we're still calling`getInstanceOrDefault()` (but ignore if it fails)

Yes that's the same behavior in 0.8.2 when default instance falls back to a docker-container one if not available (https://github.com/docker/buildx/pull/1129#issuecomment-1195542481) and therefore silently fails but still prints with bake.

But agree that we could just skip returning an instance if not required.

Reply:
@thaJeztah Should be good